test/topology/util.py

```diff
@@ -218,49 +218,60 @@ async def wait_until_last_generation_is_in_use(cql: Session):
     else:
         logger.info(f"The last generation is already in use.")
 
-async def check_system_topology_and_cdc_generations_v3_consistency(manager: ManagerClient, hosts: list[Host], cqls: Optional[list[Session]] = None):
+async def check_system_topology_and_cdc_generations_v3_consistency(manager: ManagerClient, live_hosts: list[Host], cqls: Optional[list[Session]] = None, ignored_hosts: List[Host] = []):
     # The cqls parameter is a temporary workaround for testing the recovery mode in the presence of live zero-token
     # nodes. A zero-token node requires a different cql session not to be ignored by the driver because of empty tokens
     # in the system.peers table.
-    assert len(hosts) != 0
+    assert len(live_hosts) != 0
+
+    logging.info(f"Nodes that will be ignored by check_system_topology_and_cdc_generations_v3_consistency: {ignored_hosts}")
 
     if cqls is None:
-        cqls = [manager.cql] * len(hosts)
+        cqls = [manager.cql] * len(live_hosts)
+
+    live_host_ids = frozenset(host.host_id for host in live_hosts)
+    ignored_host_ids = ignored_hosts and frozenset(host.host_id for host in ignored_hosts)
```

Comment:
Why `ignored_hosts and`?

This logic will:
- return `[]` if `ignored_hosts` is `[]`
- return a `frozenset` otherwise

is it wrong to use an empty `frozenset` in the first case?

Reply:
Removed the redundant `ignored_hosts and`